contrib/verifybinaries/verify.py

```diff
@@ -59,124 +194,457 @@ def download_with_wget(remote_file, local_file=None):
+    # binary verification?
+    trusted_keys = set()
+    if not args.no_builder_keys:
+        trusted_keys |= set(builder_key_map.keys())
+    if args.trusted_keys:
+        trusted_keys |= set(args.trusted_keys.split(','))
+
+    # Tally signatures and make sure we have enough good, trusted to fulfill
+    # our threshold.
+    good_trusted = {sig for sig in good if sig.trusted or sig.key in trusted_keys}
+    good_untrusted = [sig for sig in good if sig not in good_trusted]
+    num_trusted = len(good_trusted)
+    log.info(f"got {num_trusted} good, trusted signatures")
+
+    if num_trusted < min_trusted_sigs:
+        log.error(
+            "not enough trusted sigs to meet threshold "
+            f"({num_trusted} vs. {min_trusted_sigs})")
+        return ReturnCode.NOT_ENOUGH_GOOD_SIGS
+
+    for sig in good_trusted:
+        log.info(f"GOOD SIGNATURE: {sig}")
+
+    for sig in good_untrusted:
+        log.info(f"GOOD SIGNATURE (untrusted): {sig}")
+
+    for sig in [sig for sig in good if sig.status == 'expired']:
+        log.warning(f"key {sig.key} for {sig.name} is expired")
+
+    for sig in bad:
+        log.warning(f"BAD SIGNATURE: {sig}")
+
+    for sig in unknown:
+        log.warning(f"UNKNOWN SIGNATURE: {sig}")
 
     # extract hashes/filenames of binaries to verify from hash file;
     # each line has the following format: "<hash> <binary_filename>"
-    with open(HASHFILE, 'r', encoding='utf8') as hash_file:
-        hashes_to_verify = [
-            line.split()[:2] for line in hash_file if os_filter in line]
-    remove_files([HASHFILE])
+    with open(SUMS_FILENAME, 'r', encoding='utf8') as hash_file:
+        hashes_to_verify = [line.split()[:2] for line in hash_file if os_filter in line]
+    remove_files([SUMS_FILENAME])
     if not hashes_to_verify:
-        print("error: no files matched the platform specified")
-        return 7
+        log.error("no files matched the platform specified")
+        return ReturnCode.NO_BINARIES_MATCH
+
+    # remove binaries that are known not to be hosted by bitcoincore.org
+    fragments_to_remove = ['-unsigned', '-debug', '-codesignatures']
+    for fragment in fragments_to_remove:
+        nobinaries = [i for i in hashes_to_verify if fragment in i[1]]
+        if nobinaries:
+            remove_str = ', '.join(i[1] for i in nobinaries)
+            log.info(
+                f"removing *{fragment} binaries ({remove_str}) from verification "
+                f"since {HOST1} does not host *{fragment} binaries")
+            hashes_to_verify = [i for i in hashes_to_verify if fragment not in i[1]]
 
     # download binaries
     for _, binary_filename in hashes_to_verify:
-        print(f"Downloading {binary_filename}")
-        download_with_wget(HOST1 + remote_dir + binary_filename)
+        log.info(f"downloading {binary_filename}")
+        success, output = download_with_wget(HOST1 + remote_dir + binary_filename)
```

Comment:
This is calling wget without `-O` so if wget decides to write a .1, .2 file then then code below will verify the old existing file not the new downloaded file. Would be better to pass `binary_filename` as an explicit second argument to `download_with_wget`

Reply:
Good catch!